fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -17,166 +16,320 @@ import androidx.compose.material.IconButton
- */
-object NavigationItems {
-    val home = ActionItem(
-        iconId = R.drawable.mozac_ic_home_24,
-        descriptionResourceId = R.string.browser_toolbar_home,
-    )
+@Composable
+private fun BackButton(
+    onBackButtonClick: () -> Unit,
+    onBackButtonLongPress: () -> Unit,
+) {
+    LongPressIconButton(
+        onClick = { onBackButtonClick() },
+        onLongClick = { onBackButtonLongPress() },
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_back_24),
+            stringResource(id = R.string.browser_menu_back),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
 
-    val menu = ActionItem(
-        iconId = R.drawable.mozac_ic_ellipsis_vertical_24,
-        descriptionResourceId = R.string.mozac_browser_menu_button,
-        type = ItemType.MENU,
-    )
+@Composable
+private fun ForwardButton(
+    onForwardButtonClick: () -> Unit,
+    onForwardButtonLongPress: () -> Unit,
+) {
+    LongPressIconButton(
+        onClick = { onForwardButtonClick() },
+        onLongClick = { onForwardButtonLongPress() },
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_forward_24),
+            stringResource(id = R.string.browser_menu_forward),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
 
-    val back = ActionItem(
-        iconId = R.drawable.mozac_ic_back_24,
-        descriptionResourceId = R.string.browser_menu_back,
+@Composable
+private fun HomeButton(
+    onHomeButtonClick: () -> Unit,
+) {
+    IconButton(
+        onClick = { onHomeButtonClick() },
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_home_24),
+            stringResource(id = R.string.browser_toolbar_home),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
+
+@Composable
+private fun SearchWebButton(
+    onSearchButtonClick: () -> Unit,
+) {
+    IconButton(
+        onClick = { onSearchButtonClick() },
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_search_24),
+            stringResource(id = R.string.search_hint),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
+
+@Composable
+private fun MenuButton(
+    menuButton: MenuButton,
+) {
+    AndroidView(
+        modifier = Modifier.size(48.dp),
+        factory = { _ -> menuButton },
     )
+}
+
+@Composable
+private fun TabsButton(
+    onTabsButtonClick: () -> Unit,
+    tabCount: Int,
+) {
+    CompositionLocalProvider(LocalContentColor provides FirefoxTheme.colors.iconPrimary) {
+        IconButton(onClick = { onTabsButtonClick() }) {
+            TabCounter(tabCount = tabCount)
+        }
+    }
+}
 
-    val forward = ActionItem(
-        iconId = R.drawable.mozac_ic_forward_24,
-        descriptionResourceId = R.string.browser_menu_forward,
+@Composable
+private fun HomeNavBarPreviewRoot(isPrivateMode: Boolean) {
```

Comment:
the preview root is an older pattern for handling preview permutations. If you have the time and are curious, I'd look into using preview parameters instead. There are examples of this in `TabStrip` and `TabTools`

Reply:
Still I think I need some form of a wrapper, for providing the context and building a `MenuButton`. This preview doesn't render :(

```
@Preview
@Composable
private fun Test() {
    val context = LocalContext.current
    val menuButton = MenuButton(context)

    BrowserNavBar(
        onBackButtonClick = {},
        onBackButtonLongPress = {},
        onForwardButtonClick = {},
        onForwardButtonLongPress = {},
        onHomeButtonClick = {},
        menuButton = menuButton,
        onTabsButtonClick = {},
        isPrivateMode = false,
        browserStore = BrowserStore(),
    )
} 
```